### ISSUE TITLE
Creating static type checks for mypy

### ISSUE THREAD

**Issue by mdaehnert** [aws-cloudformation/custom-resource-helper#32]
Hi there,

I recently created a custom lambda resource where I used [mypy](https://mypy.readthedocs.io/en/stable/introduction.html) for static type checking. This tool helps a lot in maintaining clean code during bigger python projects.
mypy also checks imports if calls are used right. Right now it stops with the error message.

```sh
src\app\index.py:12: error: Cannot find implementation or library stub for module named 'crhelper'
```
Workaround is to add an ignore comment for mypy:
```python
from crhelper import CfnResource # type: ignore
```

I figured out how to easily it is to create stubs (type hints) for this project to be fully mypy compatible.

In a nutshell:
* Install _mypy_
* run _stubgen ./_ - it created *.pyi files
* add empty _py.typed_ file to this repository

If you're interested in supporting typed projects as well, then I can propose a PR with the solution mentioned above.

Best regards
Michael

**Comment by mdaehnert** [comment]
I did the typing. (Creating it was actually faster than talking about it. :-) )
Feel free to give feedback and/or merging it.

**Comment by mdaehnert** [comment]
Merged.
